syscalls/src/lib.rs

```diff
@@ -310,7 +310,7 @@ macro_rules! register_feature_gated_function {
     };
 }
 
-pub(crate) fn morph_into_deployment_environment_v1(
+pub fn morph_into_deployment_environment_v1(
```

Comment:
Can be done in a follow-up, but it might make more sense to only have `create_program_runtime_environment_v1()`, but with a parameter which controls if it is for new deployments only.